### DISCUSSION TITLE
Deploying to AWS

### DISCUSSION THREAD

**Discussion by EtienneCClarke** [vikejs/vike#883] [Discussion] [answered]
Hi, I'm trying to understand how to deploy my project. I have just migrated an SPA created with React into the vite-plugin-ssr scaffold project. Everything works as expected when starting a local NodeJS server with "npm run dev". I have attempted to host it using AWS Lambda and the Serverless Framework but seem to be hitting a load of brick walls. Anybody got some suggestions and/or know of a guide to take me through this process.

**Comment by brillout** [comment]
https://vite-plugin-ssr.com/aws-lambda has an example, and see the on-going discussion at https://github.com/brillout/vite-plugin-ssr/discussions/789 with @Strydom.

I'm happy to assist about the vite-plugin-ssr side of things.

Contribution welcome to improve the existing example or create a new one.

**Comment by EtienneCClarke** [comment]
Thanks for the speedy reply. I ended up just spinning up an EC2 Instance on AWS. But, I also keep encountering an issue when I do "npm run prod" . It shows an error with getDocumentProps not being a function. Any ideas?

`TypeError: getDocumentProps is not a function
    at render (file:///D:/.../.../.../dist/server/entries/renderer_default-page-server.mjs:33:7)`

**Reply by brillout** [reply]
Seems like a user-land error: VPS doesn't define any `getDocumentProps()`.

**Reply by EtienneCClarke** [reply]
> Seems like a user-land error: VPS doesn't define any `getDocumentProps()`.

I use it to dynamically change the head on my pages. As shown in the docs here: https://vite-plugin-ssr.com/head#by-page-dynamic. It works as expected in the dev environment just not production.

**Reply by brillout** [reply, marked as answer]
See https://github.com/brillout/vite-plugin-ssr/commit/f5e1177e48824300a136085b8d2bcc3623ecf13c.

**Reply by charlieforward9** [reply]
> I ended up just spinning up an EC2 Instance on AWS.

Did you enable HTTPS with this deployment? I am working through this right now and any pointers/snippets would be appreciated.

**Reply by EtienneCClarke** [reply]
@charlieforward9 Yes I did, using NginX as a reverse proxy and LetsEncrypt for my SSL certificate. This article on [dev.to](https://dev.to/ranjan/deploy-multiple-nodejs-apps-on-single-server-with-ssl-nginx-pm2-part-1-4841) basically goes through it step by step for you.
